Source/carthage/Build.swift

```diff
@@ -126,21 +126,100 @@ public struct BuildCommand: CommandType {
 
 public struct BuildOptions: OptionsType {
 	public let configuration: String
+	public let buildPlatform: BuildPlatform
 	public let skipCurrent: Bool
 	public let colorOptions: ColorOptions
 	public let verbose: Bool
 	public let directoryPath: String
 
-	public static func create(configuration: String)(skipCurrent: Bool)(colorOptions: ColorOptions)(verbose: Bool)(directoryPath: String) -> BuildOptions {
-		return self(configuration: configuration, skipCurrent: skipCurrent, colorOptions: colorOptions, verbose: verbose, directoryPath: directoryPath)
+	public static func create(configuration: String)(buildPlatform: BuildPlatform)(skipCurrent: Bool)(colorOptions: ColorOptions)(verbose: Bool)(directoryPath: String) -> BuildOptions {
+		return self(configuration: configuration, buildPlatform: buildPlatform, skipCurrent: skipCurrent, colorOptions: colorOptions, verbose: verbose, directoryPath: directoryPath)
 	}
 
 	public static func evaluate(m: CommandMode) -> Result<BuildOptions> {
 		return create
 			<*> m <| Option(key: "configuration", defaultValue: "Release", usage: "the Xcode configuration to build")
+			<*> m <| Option(key: "platform", defaultValue: .All, usage: "the platform to build for (one of ‘all’, ‘Mac’, or ‘iOS’)")
 			<*> m <| Option(key: "skip-current", defaultValue: true, usage: "don't skip building the Carthage project (in addition to its dependencies)")
 			<*> ColorOptions.evaluate(m)
 			<*> m <| Option(key: "verbose", defaultValue: false, usage: "print xcodebuild output inline")
 			<*> m <| Option(defaultValue: NSFileManager.defaultManager().currentDirectoryPath, usage: "the directory containing the Carthage project")
 	}
 }
+
+/// Represents the user’s chosen platform to build for.
+public enum BuildPlatform: Equatable {
+	/// Build for all available platforms.
+	case All
+
+	/// Build only for iOS.
+	case iOS
+
+	/// Build only for OS X.
+	case Mac
+
+	/// The `Platform` corresponding to this setting.
+	public var platform: Platform? {
+		switch self {
+		case .All:
+			return nil
+
+		case .iOS:
+			return .iOS
+
+		case .Mac:
+			return .Mac
+		}
+	}
+}
+
+public func == (lhs: BuildPlatform, rhs: BuildPlatform) -> Bool {
+	switch (lhs, rhs) {
+	case (.All, .All):
+		return true
+
+	case (.iOS, .iOS):
+		return true
+
+	case (.Mac, .Mac):
+		return true
+
+	default:
+		return false
+	}
+}
+
+extension BuildPlatform: Printable {
+	public var description: String {
+		switch self {
+		case .All:
+			return "all"
+
+		case .iOS:
+			return "iOS"
+
+		case .Mac:
+			return "Mac"
+		}
+	}
+}
+
+extension BuildPlatform: ArgumentType {
+	public static let name = "platform"
+
+	private static let acceptedStrings: [String: BuildPlatform] = [
+		"Mac": .Mac, "macosx": .Mac,
+		"iOS": .iOS, "iphoneos": .iOS, "iphonesimulator": .iOS,
```

Comment:
@robrix  `carthage update` almost never completes "successfully" because of code signing issues.  --platform cuts the chance of "failed builds" in half

https://www.google.com/search?q=carthage+code+signing&oq=carthage+code+signing&aqs=chrome..69i57j69i60l2.4609j0j1&sourceid=chrome&es_sm=91&ie=UTF-8

https://github.com/Carthage/Carthage/issues/281

Imagine a user who downloads xcode, creates a `Cartfile` adds `SwiftyJSON` and runs `carthage update`.... it's not a good first experience.